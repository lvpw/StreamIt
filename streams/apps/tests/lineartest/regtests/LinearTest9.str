/** Program to test the infrastructure of the linear analysis routines for pipeline combination. **/
void->void pipeline LinearTest9 {
  add FloatSource();
  add LinearPipe();
  add FloatSink();
}

void->float stateful filter FloatSource {
  float x;
  init {
    x = 0;
  }
  work push 1 {
    push(x);
    x = x + 1;
  }
}

float->void filter FloatSink {
  work pop 1 {
    println(pop());
  }
}

float->float filter FloatIdentity {
  work pop 1 push 1 {
    push(pop());
  }
}

// Start of real filters

float->float filter LinearFilter {
  work pop 3 push 1 {
    float t1 = pop();
    float t2 = pop();
    float t3 = pop();
    push(t1+t2+t3);
  }
}

float->float pipeline LinearPipe {
  add LinearFilter();
  add FloatIdentity();
}
