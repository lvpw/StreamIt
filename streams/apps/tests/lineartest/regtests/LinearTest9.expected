FloatIdentity_1.matrix
[[1.0+0.0i]]
FloatIdentity_1.vector
[[0.0+0.0i]]
LinearFilter_1.matrix
[[1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]]
LinearFilter_1.vector
[[0.0+0.0i]]
LinearPipe_1.matrix
[[1.0+0.0i]
 [1.0+0.0i]
 [1.0+0.0i]]
LinearPipe_1.vector
[[0.0+0.0i]]
